doc/cookbook/db-mysql-basics/MysqlBasics.lhs

```diff
@@ -0,0 +1,189 @@
+                                               persistLowerCase, share,
+                                               sqlSettings)
+import           Database.Persist.Types       (PersistValue(PersistInt64))
+
+import           GHC.Generics
+import           Network.Wai
+import           Network.Wai.Handler.Warp
+import           Servant
+import           Servant.API
+import           System.Environment           (getArgs)
+
+share [mkPersist sqlSettings, mkMigrate "migrateAll"] [persistLowerCase|
+Person json
+    Id   Int Primary Unique
+    name Text
+    age  Text
+    deriving Eq Show Generic
+|]
+
+data Error = Error404 | Error503 deriving (Generic, Show)
+
+data ApiResponse =
+    ApiResponseSuccessful Person
+  | ApiResponseUnsucessful Error
+  deriving (Generic, Show)
+
+instance ToJSON ApiResponse
+instance ToJSON Error
+
+type Api =
+       "person" :> Get '[JSON] [Person]
+  :<|> "person" :> Capture "id" Integer :> Get '[JSON] ApiResponse
+  :<|> "person" :> Capture "id" Integer :> Delete '[JSON] ()
+  :<|> "person" :> ReqBody '[JSON] Person :> Post '[JSON] ApiResponse
+
+apiProxy :: Proxy Api
+apiProxy = Proxy
+
+app :: Application
+app = serve apiProxy server
+
+runDB :: SqlPersistT (ResourceT (NoLoggingT IO)) a -> IO a
+runDB = runNoLoggingT . runResourceT . withMySQLConn connInfo . runSqlConn
+
+-- Change these out to suit your local setup
+connInfo :: ConnectInfo
+connInfo = defaultConnectInfo { connectHost = "127.0.0.1", connectUser = "root", connectPassword = "abcd1234", connectDatabase = "deals-api" }
+
+doMigration :: IO ()
+doMigration = runNoLoggingT $ runResourceT $ withMySQLConn connInfo $ runReaderT $ runMigration migrateAll
+
+server :: Server Api
+server = 
+  personGET :<|> 
+  personGETById :<|> 
+  personDELETE :<|>
+  personPOST
+    where
+        personGET = liftIO selectPersons
+        personGETById id = liftIO $ selectPersonById id
+        personDELETE id = liftIO $ deletePerson id
+        personPOST personJson = liftIO $ createPerson personJson
+
+selectPersons :: IO [Person]
+selectPersons = do
+  personList <- runDB $ selectList [] []
+  return $ map (\(Entity _ u) -> u) personList
+
+selectPersonById :: Integer -> IO ApiResponse
+selectPersonById id = do
+  sqlResult <- runDB $ get $ PersonKey (fromIntegral id :: Int)
+  return $ case sqlResult of
+    Just person -> ApiResponseSuccessful person
+    Nothing -> ApiResponseUnsucessful Error404
+
+createPerson :: Person -> IO ApiResponse
+createPerson person = do
+  attemptCreate <- runDB $ insert $ person
+  return $ case attemptCreate of
+    PersonKey k -> ApiResponseSuccessful person
+
+deletePerson :: Integer -> IO ()
+deletePerson id = do runDB $ delete $ PersonKey (fromIntegral id :: Int)
+
+startApp :: IO ()
+startApp = do
+    args <- getArgs
+    let arg1 = if not (null args) then Just (head args) else Nothing
+    case arg1 of
+        Just "migrate" -> doMigration
+        _              -> run 8080 app
+```
```

Comment:
Do you think you could include at the very end a few `curl` commands that hit this web application? We often do this in cookbook recipes, I think it's a nice touch.

(I can add that myself in a subsequent patch though.)

Reply:
Definitely, will add that in now.